app/controllers/frontend_log_controller.rb

```diff
@@ -10,6 +10,7 @@ class FrontendLogController < ApplicationController
 
   # rubocop:disable Layout/LineLength
   EVENT_MAP = {
+    'Frontend Error' => :frontend_error,
```

Comment:
If we want to direct all our errors to NewRelic, we could consider "noticing" (`NewRelic::Agent.notice_error`) these on the back-end?

Reply:
Let's look at the rough count/size of these from just logs and see if they're actionable before reporting? My worries are that the errors would be unactionable (we'd have to convert them to have relevant stacktraces and class names right?) and also that we just get a large large amount of them

Reply:
FWIW I'm thinking to get the errors reported nicely to `notice_error` we'd need some sort of like...

```ruby
MEMOIZED_CLASSES = Hash.new do |h, class_name|
  h[class_name] = Class.new(StandardError) do
    attr_reader :backtrace
    def initialize(message, backtrace)
      super(message)
      @backtrace = backtrace
    end
  end.tap do |klass|
    klass.define_singleton_method(:name) { "WrappedJavascript::#{class_name}" }
  end
end

# to instantiate:

err = MEMOIZED_CLASSES["TypeError"].new("message", ["backtrace:1.js"])
err.class.name
# => "WrappedJavascript::TypeError"
```

Reply:
On the point of actionability, another thing I'd like to explore here is to filter out errors happening from anything except our own JavaScript packs, to hopefully exclude most of what we're seeing in our current browsers error dashboard originating from browser extensions and app webview scripts.

Reply:
Ah yes if we filtered to only our own codebase that would definitely be more actionable

another thought is, these exceptions are different than our normal event log events, should we have a separate controller just for them? A lot of the code might end up the same, just wondering

Reply:
Yeah, I wondered about that too. I think if we went the route of funneling these to NewRelic, there's a stronger case that it should be its own controller. If we were to just log them as any other CloudWatch event, I don't know that we need a separate controller, or at least I don't know in what ways they're different.†

† with the expectation being that "filtering" would probably apply client-side, before the events are sent to the server

Reply:
Yes yes, I think we're on the same page, I meant more about "if we're reporting to NewRelic, that's different enough to warrant a new controller, maybe"

Filtering client-side seems fine, I just don't know if we know enough to validate that yet? I'm in the camp of "we don't know what we don't know" since to my knowledge we haven't worked with these errors yet.

Reply:
In e11f61a I added a branching logic path to push these errors to NewRelic.

Thoughts:

- I already don't love how we populate the frontend logger URL into the base template for every page, so I wasn't keen to the idea of adding another value for a separate controller route
- Most of the logic is shared anyways (in terms of validation and response format), where the key difference is keying off the event name and what to do with the payload

Reply:
I must have previously missed your comment at https://github.com/18F/identity-idp/pull/8950#discussion_r1286541040 which could have some benefit in terms of representing the backtrace? Not sure how necessary it is, vs. filing these away in `custom_params`.

Reply:
ok! I see we're just creating a blank `FrontendError.new` and not populating the `backtrace`, is that something we'd consider for a v2?

Reply:
@zachmargolis and I synced on Slack:

* Backtrace would be better, since it'd show as better / natively formatted within NewRelic
* Probably won't need it for now? Since we're probably (hopefully) not acting on these types of errors on a regular basis, we can put up with a bit of poor formatting when it comes time to debug
* Limiting complexity for now, but we can follow-up if we want to improve